renewal/src/renewal.cpp

```diff
@@ -55,8 +55,12 @@ int krb_ticket_renew_handler( creds_fetcher::Daemon cf_daemon )
                     std::string krb_cc_name = krb_ticket->krb_file_path;
                     std::string domainless_user = krb_ticket->domainless_user;
                     // check if the ticket is ready for renewal and not created in domainless mode
-                    if ( is_ticket_ready_for_renewal( krb_cc_name ) && domainless_user == "")
+                    if ( domainless_user == "" && is_ticket_ready_for_renewal( krb_ticket ))
```

Comment:
You can change domainless_user == "" with domainless_user.empty()
https://cplusplus.com/reference/string/string/empty/

Reply:
Addressed in latest updated PR